akka-actor/src/main/scala/akka/pattern/BackoffOnRestartSupervisor.scala

```diff
@@ -45,6 +38,13 @@ private class BackoffOnRestartSupervisor(
         // Whatever the final Directive is, we will translate all Restarts
         // to our own Restarts, which involves stopping the child.
         case Restart ⇒
+          if (strategy.withinTimeRange.isFinite() && restartCount == 0) {
+            // If the user has defined a time range for the maxNrOfRetries, we'll schedule a message
+            // to ourselves every time that range elapses, to reset the restart counter. We hide it
+            // behind this conditional to avoid queuing the message unnecessarily
+            val finiteWithinTimeRange = strategy.withinTimeRange.asInstanceOf[FiniteDuration]
+            system.scheduler.schedule(finiteWithinTimeRange, finiteWithinTimeRange, self, ResetRestartCount(restartCount))
```

Comment:
`scheduleOnce` rather, as it will then set `restartCount` to `0` and then trigger this again if needed